idaes/models_extra/column_models/MEAsolvent_column.py

```diff
@@ -0,0 +1,1483 @@
+            
+        blk.calculate_scaling_factors_unit_model()
+        iscale.calculate_scaling_factors(blk)
+        
+        # ---------------------------------------------------------------------
+        init_log.info('Step 6: Velocity constraints')
+        blk.velocity_vap.unfix()
+        blk.velocity_liq.unfix()
+        
+        for c in velocity_constraints:
+            getattr(blk, c).activate()
+            
+        use_idaes_solver_configuration_defaults()
+        optarg = {
+                "tol": 2e-8,
+                "max_iter": 150,
+                "bound_push":1e-10}
+        opt.options = optarg
+        opt.solve(blk, tee=True)
+        
+        # ---------------------------------------------------------------------
+        init_log.info('Step 7: Interfacial area constraint')
+        init_log.info('Initializing interfacial area - degrees_of_freedom = {}'.format(degrees_of_freedom(blk)))
+        blk.area_interfacial.unfix()
+        
+        for c in interfacial_area_constraint:
+            getattr(blk, c).activate()
+            
+        opt.solve(blk, tee=True)
+        
+        # Scale variable
+        for (t, x), v in blk.area_interfacial.items():
+            iscale.set_scaling_factor(v, 1/value(blk.area_interfacial[t, x]))
+            
+        # Scale constraint
+        for (t, x), v in blk.area_interfacial_constraint.items():
+            iscale.constraint_scaling_transform(
+            v, iscale.get_scaling_factor(blk.area_interfacial[t, x]))
+            
+        # ---------------------------------------------------------------------
+        init_log.info('Step 8: Liquid holdup constraint')
+        init_log.info('Initializing liquid holdup- degrees_of_freedom = {}'.format(degrees_of_freedom(blk)))
+        blk.holdup_liq.unfix()
+        
+        for c in liquid_holdup_constraint:
+            getattr(blk, c).activate()
+        
+        for x in blk.liquid_phase.length_domain:
+            if x == blk.liquid_phase.length_domain.last():
+                pass
+            else:
+                calculate_variable_from_constraint(blk.holdup_liq[0, x],
+                                                   blk.holdup_liq_constraint[0, x])
+            
+        opt.solve(blk, tee=True)
+        
+        # Scale variable
+        for (t, x), v in blk.holdup_liq.items():
+            iscale.set_scaling_factor(v, 1/value(blk.holdup_liq[t, x]))
+            
+        # Scale constraint
+        for (t, x), v in blk.holdup_liq_constraint.items():
+            iscale.constraint_scaling_transform(
+            v, iscale.get_scaling_factor(blk.holdup_liq[t, x]))
+        
+        # ---------------------------------------------------------------------    
+        init_log.info('Step 9: Vapor phase mass transfer coefficient constraint')
+        init_log.info('Initializing mass transfer coefficient (vapor phase) - degrees_of_freedom = {}'.format(degrees_of_freedom(blk)))
```

Comment:
Should we be logging the degrees of freedom, or doing a check and raising and exception if not 0?